Test/sequences/intseq_datatype.bpl

```diff
@@ -0,0 +1,57 @@
+// RUN: %boogie "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+type {:datatype} Value;
+
+function {:constructor} Integer(i: int): Value;
+function {:constructor} Vector(v: ValueArray): Value;
+
+type {:builtindecl ""} {:builtin "(Seq T@Value)"} ValueSeq;
```

Comment:
What is the empty string for?

Reply:
This is to tell Boogie to use the provided string in place of its usual declaration for this type.  In this case, we don't want any declaration.  Without this, Boogie produces:
(declare-sort (Seq T@Value))
which of course makes no sense to z3.

Reply:
I see.  The parameterized sort Seq is "pre-declared".  So Z3 will scream if we try to re-declare it.  Please see my comment above that "builtindecl" itself might be redundant.

Reply:
I would have attempted something like the following declaration for ValueSeq.

type {:builtin "Seq", "Value"} ValueSeq;

or in general

type {:builtin "SortName", "SortArg0", "SortArg1", ...}.

This will allow you to eliminate the "dependson" attribute as well because the dependency of ValueSeq on Value can now be calculated.